### PR TITLE
ARSSTB-317: Accessible file upload Service- Supporting Documents

### PR DESCRIPTION
**Pull request by C-Thomp**
### Type of change
- [x] Breaking change

### Description
Jira link: [ARSSTB-317](https://jira.tools.tax.service.gov.uk/browse/ARSSTB-317)

- adds UploadInProgress page for the user to land on while awaiting file upload
- checking progress on this new page executes out existing logic for file status from Upscan
- 
### Attachments or Screenshots

https://github.com/hmrc/advance-valuation-rulings-frontend/assets/77161245/27503e5d-cfc5-4d14-ad84-8b2af5540054

### CODE REVIEW

File: app/controllers/common/FileUploadHelper.scala [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    val redirectPath = getRedirectPath(draftId, isLetterOfAuthority, mode)
+    fileService.initiate(draftId, redirectPath, isLetterOfAuthority).map {
+      response =>
+        Ok(
+          supportingDocumentsView(
+            draftId = draftId,
+            upscanInitiateResponse = Some(response),
+            errorMessage = None
+          )
+        )
```
Comment (aaocs) [inline app/controllers/common/FileUploadHelper.scala:60]: Will this need to be wrapped in a boolean check (isLetterOfAuthority to allow for the letter of authority view when that story is done? I'm thinking along the lines of the showErrorPage method.

**Review by aaocs — APPROVED** [review]

File: conf/messages.cy [hunk 2]
```diff
@@ -933,4 +934,5 @@ yourApplicationHasBeenCancelled.paragraph.1 = This is because you cancelled your
 yourApplicationHasBeenCancelled.paragraph.2 = You can start a new application if you wish to do so.
 yourApplicationHasBeenCancelled.button = Go to your applications and rulings
 
-supportingDocuments.remove.hidden = remove this document from your list of supporting documents
\ No newline at end of file
+uploadInProgress.title = Upload In Progress
+uploadInProgress.heading = Uploading your document...
```
Comment (aaocs) [inline conf/messages.cy:938]: Do we need Welsh translations for these messages?

File: app/controllers/common/FileUploadHelper.scala (hunk 1 above)
Comment (aaocs) [reply app/controllers/common/FileUploadHelper.scala:60]: Answer: yes.

File: conf/messages.cy (hunk 2 above)
Comment (aaocs) [reply conf/messages.cy:938]: Answer: Yes. ARSSTB-389 has been created for this.

**Review by paulhargrave — APPROVED** [review]
:)

**Pull request merged**